### ISSUE TITLE
TransformServerTypings DateTime => string issue

### ISSUE THREAD

**Issue by fwd079** [volkanceylan/Serenity#5168]
Hi @volkankalkan 

Sorry to bother, a quick yes/no question, just noticed that TransformServerTypings is actually transforming `DateTime` or `DateTime?` columns into `string` columns at TypeScript side. 

Which means I cannot pass something like:

```typescript
let row = MyRow {
    row.UpdateDate = new Date();
}
```

Because TypeScript gives me this error:

```batch
TS2322 TypeScript (TS) Type 'Date' is not assignable to type 'string'.
```
Yes I know I can try to look at parsing etc but not something ideal when we **do** have a `Date` type.

So the question. Is this "output DateTime as string" step your TransformServerTypings tool intentional, or a bug?

Kind Regards.

**Comment by volkanceylan** [comment]
It's purely intentional, there is no Date type in JSON, so we serialize dates as ISO 8604 JSON string and editors also have string value. You may use Q.formatDate(new Date(), "yyyy-MM-dd") or Q.formatIsoDateTime(new Date())

Also never set UpdateDate etc  on client side.

**Comment by fwd079** [comment]
Good afternoon @volkanceylan,

> It's purely intentional, there is no Date type in JSON, so we serialize dates as ISO 8604 JSON string and editors also have string value. You may use Q.formatDate(new Date(), "yyyy-MM-dd") or Q.formatIsoDateTime(new Date())
> 
>  
  

Okay that is interesting. We do have a `Date` [interface (click)](https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_es5_d_.date.html) inside TypeScript, that is where I was thinking Serenity will take hint whenever a `*.d.ts` file is generated for an `*Endpoint.cs` class. And you'd take `Date` off client side, and cast it at send-to-JSON time (JIT style).

> 
> Also never set UpdateDate etc on client side.
>  
  

Yeah I know, was asking if we do have `Date` object in Serenity `Service` files. That "update date" was just to give you a simple example. There are no hard rules anywhere for not setting dates from a client side object to be sent back to any API for navigation purposes, that is where JavaScript [Date](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) object comes into play.

There are times when we do need to send a particular Date value from client side (in this case "start date of my list" value) on a page. For example in this particular case **logic is not inside any Serenity grid or dialogue.** Instead, this value is used inside a third-party control to get rows from database when a button is clicked.

Anyway, I have already solved my problem in good old `any` way. lol
Here is sample code for readers.

```typescript
        private populateDataLists(startDate: Date, thresholdDate: Date): void {

            MyService.GetDataLists({
                ColumnSelection: Serenity.ColumnSelection.Details,
                StartDate: startDate as any,
                ThresholdDate: thresholdDate as any,
            }, response => {
                this.dataList1= response.Entity.List1;
                this.dataList2= response.Entity.List2;
            }, this.serviceOptions);
        }
```

This is handled fine by Serenity actually, which is good to know, because we do have Serenity in same project. So just thought of leveraging a Serenity's `Service.Retrieve()` like method I wrote, that is where I realised there was no `Date` type inside TypeScript declaration files generated for `*Endpoint.cs` files. So thought of asking a quick yes/no question as did not like casting to `any` type. 

The answer is "We intentionally do not cast `Date` object" so will pass it on to my team, thank you for your time. Closing the question now.

Kind Regards.
